packages/frontend/app/routes/wallet-addresses.create.tsx

```diff
@@ -20,23 +20,51 @@ import {
 } from '~/shared/utils'
 import { checkAuthAndRedirect } from '../lib/kratos_checks.server'
 import { type LoaderFunctionArgs } from '@remix-run/node'
-import { whoAmI, loadTenants } from '~/lib/api/tenant.server'
+import type { listTenants } from '~/lib/api/tenant.server'
+import { whoAmI, loadTenants, getTenantInfo } from '~/lib/api/tenant.server'
+
+const WALLET_ADDRESS_URL_KEY = 'WALLET_ADDRESS_URL'
+
+const findWASetting = (
+  tenantSettings: Awaited<ReturnType<typeof getTenantInfo>>['settings']
+) => {
+  return tenantSettings.find(
+    (setting) => setting.key === WALLET_ADDRESS_URL_KEY
+  )?.value
+}
+
+const findTenant = (
+  tenants: Awaited<ReturnType<typeof listTenants>>['edges'],
+  tenantId: string
+) => {
+  return tenants.find((tenant) => tenant.node.id === tenantId)
+}
 
 export const loader = async ({ request }: LoaderFunctionArgs) => {
   const cookies = request.headers.get('cookie')
   await checkAuthAndRedirect(request.url, cookies)
 
   const assets = await loadAssets(request)
-  const { isOperator } = await whoAmI(request)
+  const { id, isOperator } = await whoAmI(request)
   let tenants
+  let tenantWAPrefix
   if (isOperator) {
-    tenants = await loadTenants(request)
+    const loadedTenants = await loadTenants(request)
+    tenants = loadedTenants.filter(
+      (tenant) => findWASetting(tenant.node.settings) || tenant.node.id === id
+    )
+  } else {
+    const tenant = await getTenantInfo(request, { id })
+    const waPrefixSetting = findWASetting(tenant.settings)
+    if (!waPrefixSetting)
+      throw new Error('Wallet Address Prefix not configured!')
```

Comment:
This shouldn't be possible, correct? Otherwise, maybe we shouldn't throw and just fallback to the open payments URL?